### DISCUSSION TITLE
Querying Records by Select field when using PB as Framework

### DISCUSSION THREAD

**Discussion by apikee** [pocketbase/pocketbase#1959] [Q&A]
Hello, is there a way to query records by Select field? I wanted to do something like this:

```
emails, err := app.Dao().FindRecordsByExpr(EMAIL_HISTORY_COLLECTION,
	dbx.HashExp{"type": []string{"marketing", "source"}},
)
```

but had to do it like this, since I wasn't able to use the string slice

```
emails, err := app.Dao().FindRecordsByExpr(EMAIL_HISTORY_COLLECTION,
        dbx.HashExp{"type": "[\"marketing\",\"source\"]"},
)
```

I'm I missing something? Thanks

**Comment by ganigeorgiev** [comment]
Is the above supposed to be like an "IN" expression or you want to check both values?


Multi-valued fields (including `select` with "Max select > 1") are stored as serialized json array.

If you want to apply a condition against each individual value from the serialized array you have to use [`json_each`](https://www.sqlite.org/json1.html#jeach).
To do that you can create a custom query, something like:

```go
collection, err := app.Dao().FindCollectionByNameOrId(EMAIL_HISTORY_COLLECTION)
if err != nil {
	return err
}

records := []*models.Record{}

err = app.Dao().
	RecordQuery(collection).
	InnerJoin("json_each(type) je", nil).
	AndWhere(dbx.In("je.value", "marketing", "source")). // example with IN, adjust depending on your use case
	All(&records)
if err != nil {
	return err
}
```

**Reply by apikee** [reply]
Thanks for the reply. I want to check both values -> I want to find records that have both "marketing" and "source" as a "type" property. Can the json_each be used as well for the usecase?

**Reply by ganigeorgiev** [reply]
Is `type` single or multiple `select` field?

**Reply by ganigeorgiev** [reply]
If it is **single** `select` and you want to get records that have `type` "marketing" or "source":

```go
emails, err := app.Dao().FindRecordsByExpr(
    EMAIL_HISTORY_COLLECTION,
    dbx.HashExp{"type": []any{"marketing", "source"}}, // notice that we are using any/interface{}, not string as slice type
)

// OR use direct IN expression

emails, err := app.Dao().FindRecordsByExpr(
    EMAIL_HISTORY_COLLECTION,
    dbx.In("type", "marketing", "source"),
)
```

---

If it is **multiple** `select` and you want to get records that have `type` field containing **both** "marketing" and "source", then `json_each` will not work in this case since it'll apply a condition for each item of the array. SQLite doesn't have built-in match-all/contains-all type of expression, but there are workaround approaches - combine the join of `json_each` with group by and/or count OR use directly a LIKE expression against the serialized string value (this is the easiest):

```go
emails, err := app.Dao().FindRecordsByExpr(
    EMAIL_HISTORY_COLLECTION,
    dbx.Like("type", `"marketing"`),
    dbx.Like("type", `"source"`),
)
```

**Reply by apikee** [reply]
Thanks for the response, the second case is what i'm looking for -> it's multiple select and I wanted to find records containing both of the types "marketing" and "source". Thank you
